Added TypeScript support

Shin-- opened the pull request:
I added `buntis` (https://github.com/buntis/buntis) for TS parsing and `estree-walker` (https://github.com/Rich-Harris/estree-walker) to walk through it.

When I was basically done, I noticed that `buntis` only lists tracking line numbers as an option, but apparently it is not implemented (https://github.com/buntis/buntis/pull/40). I didn't have luck finding another AST parser for TypeScript to replace this, so I just set the line numbers to always be 0, which leads to .po files not showing a line number for those translations. I have no experience with AST and parsing in general, so I tried to stick close to how it is done for JavaScript and `acorn`.

I kept the tests pretty much the same as the JavaScript tests except for adding some TypeScript specific syntax to some templates. This does not test TypeScript extensively, should cover most cases though. In one test (`should be able to parse correctly template strings without variables`), I have added a `\n` at the start of the string comparison. The template starts with a `\n`, the JavaScript parser would return it without the new line though. I also commented out any checks for it being the correct line.

vperron commented:
that looks nice. Could you please rebase to avoid the conflicts ? It will help us read that properly  !

Shin-- commented:
@vperron Yeah, I just noticed that, after creating the pull request. Unfortunately the latest merge removed the checks for ts/js from previous versions so I had to readd this. I read https://github.com/Polyconseil/easygettext/pull/74 + adjusted tests, so everything should still work as before. Me creating a pull request while my master was not up to date led to a lot of unnecessary commits though. Sorry for that.
Also looks like I have to improve the tests.

vperron commented:
The tests were improved and we fixed a pretty bad regression which did disallow us to extract strings from simple HTML files, which was the original goal of easygettext :) 

OK thanks I'll read that asap.

Shin-- commented:
@vperron I didn't mean that the current tests need improvement but that mine reduced the code coverage and the checks failed

vperron commented:
Ah sorry, I was talking about the commits preceding your rebase. 

About yours a code coverage of 97% is still alright, don't worry !

vperron commented on src/extract-cli.js:
```diff
@@ -67,7 +67,13 @@ files.forEach(function(filename) {
+    if (script && script.lang === 'ts' ) {
+      extractor.parseTypeScript(file, script.content);
+    } else {
+      extractor.parseJavascript(file, data);
```
it really seems to me that you should give `script` instead of `data` here. Or you'll break the whole javascript decoding feature.

Shin-- replied:
Thanks, this was unintentional

vperron commented on src/extract-cli.js:
```diff
@@ -67,7 +67,13 @@ files.forEach(function(filename) {
+    let script = extract.preprocessJavascript(data, ext)
+
+    if (script && script.lang === 'ts' ) {
+      extractor.parseTypeScript(file, script.content);
```
I'd be really glad to see you using the same type as the function below, say: `script` and not `script.content`.

If thos two functions look like they have the same signature but they don't it'll lead to bad issues later.

Shin-- replied:
I am using `script.content` in both cases now. I changed `preprocessJavascript` earlier to return an object that contains `content` and `lang`. The other option would be to make `parseTypeScript` and `parseJavascript` handle this object, but I think that would make those functions less general.

vperron commented on src/extract-cli.js:
```diff
@@ -67,7 +67,13 @@ files.forEach(function(filename) {
     let data = fs.readFileSync(file, {encoding: 'utf-8'}).toString();
     extractor.parse(file, extract.preprocessTemplate(data, ext));
-    extractor.parseJavascript(file, extract.preprocessJavascript(data, ext));
+    let script = extract.preprocessJavascript(data, ext)
```
you can probably use `const` here.

Shin-- commented:
@vperron I think I have shown how much chaos you can create with commits. Honestly, quite embarrassing how this escalated. I never squashed commits, and I think my squashed commit was wrong since I should have just forced pushed it. I hope my latest one fixes it.
Anyway, I updated the ReadMe and added support for .ts files. Before I only supported ts scripts inside .vue files. I think the latest PR that moved some of the logic to `processJavascript` broke support for non-vue files. `.js` and `.ts` should be supported now.
